nservicebus/upgrades/8to9/index.md

```diff
@@ -1,18 +1,103 @@
+
+snippet: core-8to9-machinename
+
+## API to set additional audit metadata has changed
+
+The API to [set additional audit metadata](/nservicebus/operations/auditing.md#additional-audit-information) has been changed.
+
+snippet: core-8to9-audit-metadata
+
+## Dependency registration access in features renamed
+
+The property used to access container registrations in features has been renamed from `Container` to `Services`.
+
+snippet: core-8to9-di-features
+
+## Service collection extensions for backwards compatibility removed
+
+Service collection extensions to ease [the transition to Microsoft DI abstractions](/nservicebus/upgrades/7to8/dependency-injection.md#registercomponents-changes) have been removed. It is now required to use the registration APIs added in NServiceBus 8.
+
+snippet: core-8to9-di-shims
+
+## Making features depend on message driven subscriptions
+
+The API to make features depend on [message driven subscriptions](/nservicebus/messaging/publish-subscribe/#mechanics-message-driven-persistence-based) being in use has changed.
+
+snippet: core-8to9-depend-on-subscriptions
+
+## Endpoint addresses
+
+### Access to logical addresses in features
+
+Since endpoint addresses are translated to transport-specific ones later during endpoint startup, addresses are defined using a transport-agnostic `QueueAddress` type. The addresses can be accessed via the `FeatureConfigurationContext`:
+
+snippet: core-8to9-adresses-features
+
+### Access to the physical receive addresses at runtime
+
+Inject the `ReceiveAddresses` type to access the endpoint receive addresses at runtime.
+
+snippet: core-8to9-adresses-runtime
+
+### Dynamic address translation
+
+Inject the new `ITransportAddressResolver` to translate `QueueAddress`'s to a physical transport addresses at runtime.
+
+snippet: core-8to9-adresses-translation
+
+## Restricted type argument for .UsePersistence<T>
+
+https://github.com/Particular/NServiceBus/compare/release-8.1...master#diff-2c08aef8335f8f17ba3dc362fe939f8a5bdddde4f411d06a067882ac204fa43eL854
+
+Proposed action: Exclude since it doesn't impact users
+
+## Exceptions no longer marked as serializable
+
+https://github.com/Particular/NServiceBus/compare/release-8.1...master#diff-2c08aef8335f8f17ba3dc362fe939f8a5bdddde4f411d06a067882ac204fa43eL739
+
+Proposed action: Exclude since it doesn't impact users
+
+## Extensionpoint for non task based notifications has been deprecated
+
+https://github.com/Particular/NServiceBus/compare/release-8.1...master#diff-2c08aef8335f8f17ba3dc362fe939f8a5bdddde4f411d06a067882ac204fa43eR250 (TBD remove)
+
+Propose action: Don't mention this since the only use case was already deprecated in v8, see /nservicebus/upgrades/7to8/#error-notification-events and no downstream is using this API (there could be end users extending this though but that feels very unlikely?)
+
+## Audit context no longer accept time to be received
```

Comment:
What have we done in the past for these sort of pipeline changes? If we've documented them in the past, might be a good idea to go ahead.

Reply:
Decided to not document this since creating an audit context requires a ForkConnector to be used 

https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs#L209

while a user could do that there is no way to actually invoke the fork since the `Fork` method is internal

https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/ForkExtensions.cs#L6

and inlining the `InvokePipeline` call won't work either since its internal as well

https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Pipeline/PipelineInvocationExtensions.cs#L6

TL;DR; Not documenting this since no use outside core is possible